### ISSUE TITLE
"Inline temporary variable" introduces an unnecessary cast that's reported as nullability issue

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#60552]
**Version Used**: 
17.2.0 Preview 2.0

**Steps to Reproduce**:
```cs
public class C
{
    private struct S
    {
    }

    public string M()
    {
        S s;
        var a = "" + s; // "Inline temporary variable" for a
        return a;
    }
}
```
Run "Inline temporary variable" for `a`.

**Expected Behavior**:
```cs
public class C
{
    private struct S
    {
    }

    public string M()
    {
        S s;
        // "Inline temporary variable" for a
        return "" + s;
    }
}
```

**Actual Behavior**:
```cs
public class C
{
    private struct S
    {
    }

    public string M()
    {
        S s;
        // "Inline temporary variable" for a
        return (string?)("" + s);
    }
}
```
The refactoring introduces a cast to `(string?)` even though neither `a` nor the return type were ever nullable.

On a related note: Is the comment trivia expected to stay where it is and not move to the `return` line?

**Comment by Youssef1313** [comment]
I think this is because `var` has is inferred as nullable type. A fix for this may reuse helpers in #58866

**Comment by CyrusNajmabadi** [comment]
We would take a targeted community fix here.
